### PR TITLE
Modified configuration structure and related updates

### PR DESCRIPTION
**Pull request by efratmuller**
Also added documentation

### CODE REVIEW

File: src/ml_pipeline/config.yml [hunk 1]
```diff
@@ -45,16 +37,18 @@ default:
     'feature_importance_csv': '../../data/ml_output/results_tables/%s_feature_importance.csv',
     'cv_results_csv': '../../data/ml_output/results_tables/%s_pipeline.csv'
   }
-  
+  params_short_names: {
+    'should_tune': {FALSE: '-T', TRUE: '+T'},
```
Comment (ramti) [inline src/ml_pipeline/config.yml:41]: I think it should be "false" and "true".

File: src/ml_pipeline/ml_pipeline.R [hunk 2]
```diff
@@ -352,90 +353,100 @@ ml_pipeline_single_run <- function(ds_name,
...
+    return(results)
+  })
+  
   all_results <- bind_rows(all_results)
+  
   return(tibble_row(dataset = ds_name, results = list(all_results)))
 }
 
 
 # Not in use, legacy
```
Comment (ramti) [inline src/ml_pipeline/ml_pipeline.R:421]: If it's legacy, let's remove it...

File: src/ml_pipeline/ml_pipeline.R [hunk 3]
```diff
@@ -352,90 +353,100 @@ ml_pipeline_single_run <- function(ds_name,
...
   predefined_ds_config <- NULL
-  if (!is.null(predefined_ds_config_path)) {
-    predefined_ds_config <- read_csv(predefined_ds_config_path) %>%
+  if (!is.null(predefined_config_path)) {
+    predefined_ds_config <- read_csv(predefined_config_path) %>%
       filter(dataset == ds_name) %>%
       select(-c(dataset))
+    
+    # Quit if dataset is not in predefined list
+    if (nrow(predefined_ds_config)==0) {
```
Comment (ramti) [inline src/ml_pipeline/ml_pipeline.R:446]: if (nrow(predefined_ds_config) == 0) {

File: src/ml_pipeline/ml_pipeline.R [hunk 4]
```diff
@@ -104,8 +104,10 @@ ml_create_rf_model <- function(ds_name,
 }
 
 
-ml_fit_model_on_train <-
-  function(ds_name, train_data, rf_obj, data_recipe) {
+ml_fit_model_on_train <- function(run_name,
```
[the remark this answers is not in the document]
Comment (efratmuller) [reply src/ml_pipeline/ml_pipeline.R:107]: fixed in next commit

File: src/ml_pipeline/ml_pipeline.R (hunk 2 above)
Comment (efratmuller) [reply src/ml_pipeline/ml_pipeline.R:421]: fixed in next commit

File: src/ml_pipeline/ml_pipeline.R (hunk 3 above)
Comment (efratmuller) [reply src/ml_pipeline/ml_pipeline.R:446]: ?

Comment (efratmuller) [reply src/ml_pipeline/ml_pipeline.R:446]: spaces? added

File: src/ml_pipeline/config.yml (hunk 1 above)
Comment (efratmuller) [reply src/ml_pipeline/config.yml:41]: it works like this (I think it treats it as characters not as booleans, i.e. I could also quote it

**Review by ramti — APPROVED** [review]

**Pull request merged**